lib/package-json-reader.js

```diff
@@ -16,30 +16,28 @@ export default reader
 
 /**
  * @param {string} jsonPath
- * @returns {{string: string|undefined}}
+ * @returns {{string: string|undefined,path: string|undefined}}
  */
 function read(jsonPath) {
   return find(path.dirname(jsonPath))
 }
 
 /**
  * @param {string} dir
- * @returns {{string: string|undefined}}
+ * @returns {{string: string|undefined,path: string|undefined}}
  */
 function find(dir) {
   try {
-    const string = fs.readFileSync(
-      path.toNamespacedPath(path.join(dir, 'package.json')),
-      'utf8'
-    )
-    return {string}
+    const pjsonPath = path.toNamespacedPath(path.join(dir, 'package.json'))
+    const string = fs.readFileSync(pjsonPath, 'utf8')
+    return {string, path: pjsonPath}
   } catch (error) {
     const exception = /** @type {ErrnoException} */ (error)
 
     if (exception.code === 'ENOENT') {
       const parent = path.dirname(dir)
       if (dir !== parent) return find(parent)
-      return {string: undefined}
```

Comment:
The source does not pass `pjsonPath` back, and it’s not used in resolve: https://github.com/nodejs/node/blob/40fa2e9c110c616b1301c20c2a3cdb2cf6cc165e/lib/internal/modules/esm/resolve.js#L181.

I’m wary of changing the source here in ways other than in Node.js itself, as it would be reverted by the next changes.
Could some other change be needed?

Reply:
yeah, I was also thinking it's weird that the test passes in built in `import.meta.resolve` but fails here, even though the source is the same... do you have any clue what might be the reason (without comparing line by line?)

Reply:
I guess the change should indeed go to `package-json-reader`, https://github.com/nodejs/node/blob/40fa2e9c110c616b1301c20c2a3cdb2cf6cc165e/lib/internal/modules/package_json_reader.js

Reply:
OK, I got it! Just needed to remove the find-up mechanism of `package-json-reader`